### PR TITLE
Do not discover unregistered API #5

### PR DESCRIPTION
**Pull request by rathnapandi**
Fix to discover active apis

### CODE REVIEW

**Comment by rathnapandi** [comment]
change filtering based on querystring

**Review by tsjohns9 — COMMENTED** [review]
This looks good. Can you show it to me? I will approve it after that.

File: pkg/anypoint/types.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ type API struct {
 	Pinned                    bool     `json:"pinned"`
 	ProductVersion            string   `json:"productVersion"`
 	Tags                      []string `json:"tags"`
+	LastActiveDate            string   `json:lastActiveDate`
```
Comment (tsjohns9) [inline pkg/anypoint/types.go:95]: Is this field still needed after your new changes?

File: pkg/anypoint/client.go [hunk 2]
```diff
@@ -450,15 +453,20 @@ func (c *AnypointClient) CreateContract(appID int64, contract *Contract) (*Contr
...
 	}
 
-	query := map[string]string{
-		"masterOrganizationId": c.auth.GetOrgID(),
+	if query == nil {
+		query = map[string]string{}
 	}
 
+	query["masterOrganizationId"] = c.auth.GetOrgID()
+	// query := map[string]string{
```
Comment (tsjohns9) [inline pkg/anypoint/client.go:466]: Please remove the commented code before merging

File: pkg/anypoint/types.go (hunk 1 above)
Comment (rathnapandi) [reply pkg/anypoint/types.go:95]: No, i can remove it.

File: pkg/discovery/servicehandler.go [hunk 3]
```diff
@@ -237,7 +237,13 @@ func (s *serviceHandler) createSubscriptionSchemaForSLATier(
 }
 
 // shouldDiscoverAPI determines if the API should be pushed to Central or not
-func shouldDiscoverAPI(endpoint string, discoveryTags, ignoreTags, apiTags []string) bool {
+func shouldDiscoverAPI(endpoint string, discoveryTags, ignoreTags, apiTags []string, lastActiveDate string) bool {
+
+	// If lastActiveDate is empty, API is unregistered
+	if lastActiveDate == "" {
```
[the remark this answers is not in the document]
Comment (rathnapandi) [reply pkg/discovery/servicehandler.go]: agree

**Comment by rathnapandi** [comment]
Fix review comments

**Review by tsjohns9 — APPROVED** [review]

**Pull request merged**